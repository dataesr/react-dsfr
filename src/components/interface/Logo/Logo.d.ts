import * as React from 'react';

export type LogoClassName = string | Object | any[];

export type LogoChildren = string | React.ReactNode;

export interface LogoProps {
    __TYPE?: any;
    className?: LogoClassName;
    children: LogoChildren;
    hrefTitle?: string;
    href?: string;
    /**
     * Number of characters to split title
     */
    splitCharacter?: number;
    asLink?: React.ReactElement<any>;
}

declare const Logo: React.FC<LogoProps>;

export default Logo;

