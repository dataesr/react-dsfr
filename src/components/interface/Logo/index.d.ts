import type Logo from "./Logo"

export type { Logo }
