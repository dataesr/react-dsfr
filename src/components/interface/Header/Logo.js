import classnames from 'classnames';
import PropTypes from 'prop-types';
import { CLASS_NAME_TYPE } from '../../../utils/variables';

const Logo = ({ src, alt, className }) => (
  <div className={classnames(className, 'rf-header__operator')}>
    <img src={src} alt={alt} />
  </div>
);

Logo.defaultProps = {
  className: '',
};

Logo.propTypes = {
  src: PropTypes.string.isRequired,
  alt: PropTypes.string.isRequired,
  /**
 * One of: string, object
 */
  className: CLASS_NAME_TYPE,
};
export default Logo;
