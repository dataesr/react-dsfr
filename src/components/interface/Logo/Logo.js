import React from 'react';
import PropTypes from 'prop-types';
import { v4 as uuidv4 } from 'uuid';
import classNames from 'classnames';
import Link from '../Link/index';
import dataAttributes from '../../../utils/data-attributes';
import typeValidation from '../../../utils/type-validation';

/*
* DSFR v1.0.0
*/
import '@gouvfr/dsfr/dist/css/core.min.css';
import '@gouvfr/dsfr/dist/css/content.min.css';
import '@gouvfr/dsfr/dist/css/forms.min.css';
import '@gouvfr/dsfr/dist/css/inputs.min.css';

/*
* react-ds-fr
*/
import '../../../style/colors.css';
import '../../../style/custom.css';

import '@gouvfr/dsfr/dist/css/logo.min.css';

/**
 *
 * @visibleName Logo
 */
const Logo = ({
  children, href, to, className, splitCharacter, hrefTitle, ...remainingProps
}) => {
  // TODO Better system to split
  const regex = new RegExp(`.{${splitCharacter.toString()}}\\S*\\s+`, 'g');
  const arrayStr = children.replace(regex, '$&@').split(/\s+@/);
  const title = arrayStr.reduce((el, a, i) => el.concat(a, i < arrayStr.length - 1 ? (
    <br key={uuidv4()} />) : ''), []);

  return (
    <Link
      className={classNames(className)}
      title={hrefTitle || children}
      href={href}
      to={to}
      {...dataAttributes(remainingProps)}
    >
      <p className="fr-logo">
        {title}
      </p>
    </Link>
  );
};

Logo.defaultProps = {
  __TYPE: 'Logo',
  href: undefined,
  to: undefined,
  hrefTitle: '',
  splitCharacter: 12,
  className: '',
};

Logo.propTypes = {
  // eslint-disable-next-line react/no-unused-prop-types
  __TYPE: typeValidation('Logo'),
  className: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
    PropTypes.array,
  ]),
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
  hrefTitle: PropTypes.string,
  href: PropTypes.string,
  to: PropTypes.string,
  /**
  * Number of characters to split title
  */
  splitCharacter: PropTypes.number,
};

export default Logo;
