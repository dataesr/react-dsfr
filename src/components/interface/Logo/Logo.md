## Logo - Bloc-marque
La structure du bloc-marque est commune à toutes les entités de l’État.

```jsx
<Logo>
    Ministère de l'enseignement supérieur de la recherche et de l'innovation
</Logo>
```
