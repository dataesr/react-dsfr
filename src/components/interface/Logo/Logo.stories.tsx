import { ComponentStory } from '@storybook/react';

import Logo from './Logo';

export default {
  title: 'Composants/Logo',
  component: Logo,
  parameters: {
    docs: {
      description: {
        component: 'Le composant permet de formatter le texte qui accompagne le logo du Ministère.',
      },
    },
  },
};

const Template: ComponentStory<typeof Logo> = (args) => <Logo {...args} />;

export const FormatLogo = Template.bind({});
FormatLogo.args = {
  children: 'Ministère de l\'enseignement supérieur et de la Recherche',
};
FormatLogo.storyName = 'Logo formatté';
