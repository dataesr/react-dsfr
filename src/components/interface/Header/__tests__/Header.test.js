import Enzyme from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { v4 as uuidv4 } from 'uuid';
import {
  Header, HeaderBody, HeaderOperator, Service, Tool, ToolItemGroup,
  HeaderNav, NavItem, NavSubItem, ToolItem, ToolTranslate, ToolTranslateItem,
} from '../index';
import Logo from '../../Logo';

Enzyme.configure({ adapter: new Adapter() });

jest.mock('uuid', () => ({
  v4: jest.fn(),
}));

describe('<Header />', () => {
  beforeEach(() => {
    uuidv4.mockImplementationOnce(() => 'xxxxxxx');
    uuidv4.mockImplementationOnce(() => 'yyyyyy');
  });

  it('renders correctly', () => {
    const component = renderer
      .create(
        <Header data-testid="header">
          <HeaderBody data-testid="headerbody">
            <Logo splitCharacter={10} data-test-id="headerlogo">République française</Logo>
            <HeaderOperator>
              <img src="https://fakeimg.pl/145x81/" alt="texte alternatif" />
            </HeaderOperator>
            <Service
              title="Nom du service"
              description="baseline - précisions sur l‘organisation"
              data-testid="service"
            />
            <Tool
              closeButtonLabel="fermer"
              data-testid="tool"
            >
              <ToolItemGroup data-testid="toolitemgroup">
                <ToolItem icon="fr-fi-lock-line" link="/path" data-testid="toolitem">Se connecter</ToolItem>
                <ToolItem icon="fr-fi-add-circle-line" link="/path" data-test-id="toolitem">Créer un espace</ToolItem>
                <ToolTranslate currentLang="FR" descCurrentLang="Français">
                  <ToolTranslateItem href="/fr" hrefLang="FR" active>
                    Français
                  </ToolTranslateItem>
                  <ToolTranslateItem href="/en" hrefLang="EN" active={false}>
                    English
                  </ToolTranslateItem>
                </ToolTranslate>
              </ToolItemGroup>
            </Tool>
          </HeaderBody>
          <HeaderNav data-testid="headernav">
            <NavItem title="Home" link="/" data-testid="navitem" />
            <NavItem title="Resources" data-test-id="navitem">
              <NavSubItem title="Ressource #1" link="/path-to-resources-1" data-test-id="navsubitem" />
              <NavSubItem title="Ressource #2" link="/path-to-resources-2" data-test--id="navsubitem" />
              <NavSubItem title="Ressource #3" link="/path-to-resources-3" data-test="navsubitem" />
            </NavItem>
          </HeaderNav>
        </Header>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
