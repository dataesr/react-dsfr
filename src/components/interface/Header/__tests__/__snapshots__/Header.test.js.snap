// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
<header
  className="fr-header"
  data-testid="header"
  role="banner"
>
  <div
    className="fr-header__body"
    data-testid="headerbody"
  >
    <div
      className="fr-container"
    >
      <div
        className="fr-header__body-row"
      >
        <div
          className="fr-header__brand fr-enlarge-link"
        >
          <div
            className="fr-header__brand-top"
          >
            <div
              className="fr-header__logo"
            >
              <a
                className=""
                data-test-id="headerlogo"
                target="_self"
                title="République française"
              >
                <p
                  className="fr-logo"
                >
                  République
                  <br />
                  française
                  
                </p>
              </a>
            </div>
            <div
              className="fr-header__operator"
            >
              <img
                alt="texte alternatif"
                src="https://fakeimg.pl/145x81/"
              />
            </div>
            <div
              className="fr-header__navbar"
            >
              <button
                aria-label="ouvrir la navigation"
                className="fr-btn--menu fr-btn"
                onClick={[Function]}
                type="button"
              />
            </div>
          </div>
          <div
            className="fr-header__service"
            data-testid="service"
          >
            <a
              className="fr-header__service-title"
              target="_self"
              title="Nom du service"
            >
              Nom du service
            </a>
            <p
              className="fr-header__service-tagline"
            >
              baseline - précisions sur l‘organisation
            </p>
          </div>
        </div>
        <div
          className="fr-header__tools"
          data-testid="tool"
        >
          <div
            className="fr-header__tools-links"
            data-testid="toolitemgroup"
          >
            <ul
              className="fr-links-group"
            >
              <li
                data-testid="toolitem"
              >
                <a
                  className="ds-fr--flex fr-link"
                  target="_self"
                >
                  <span
                    className="ri-1x icon-left fr-fi-lock-line ds-fr--v-top"
                  />
                  Se connecter
                </a>
              </li>
              <li
                data-test-id="toolitem"
              >
                <a
                  className="ds-fr--flex fr-link"
                  target="_self"
                >
                  <span
                    className="ri-1x icon-left fr-fi-add-circle-line ds-fr--v-top"
                  />
                  Créer un espace
                </a>
              </li>
            </ul>
          </div>
          <div
            className="fr-header__search fr-modal"
          >
            <div
              className="fr-container fr-container-lg--fluid"
            >
              <button
                aria-controls="header-tools-popin"
                className="fr-link--close fr-link"
                onClick={[Function]}
                title="Fermer"
                type="button"
              >
                Fermer
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="fr-header__menu fr-modal"
    data-testid="headernav"
  >
    <div
      className="fr-container"
    >
      <button
        aria-label="fermer la navigation"
        className="fr-link--close fr-link"
        onClick={[Function]}
        title="Fermer"
        type="button"
      >
        Fermer
      </button>
      <nav
        aria-label="Menu principal"
        className="fr-nav"
        id="header-navigation"
        role="navigation"
      >
        <ul
          className="fr-nav__list"
        >
          <li
            className="fr-nav__item"
          >
            <a
              className="fr-nav__link"
              target="_self"
            >
              Home
            </a>
          </li>
          <li
            className="fr-nav__item"
            data-test-id="navitem"
          >
            <button
              aria-controls="fr-nav-subitem-undefined"
              aria-expanded={false}
              aria-label="ouvrir la navigation"
              className="fr-nav__btn"
              onClick={[Function]}
              type="button"
            >
              Resources
            </button>
            <div
              className="fr-menu fr-collapse"
              id="fr-nav-subitem-undefined"
              style={
                Object {
                  "--collapse": "0px",
                  "maxHeight": null,
                }
              }
            >
              <ul
                className="fr-menu__list"
              >
                <li
                  data-test-id="navsubitem"
                >
                  <a
                    className="fr-nav__link"
                    target="_self"
                  >
                    Ressource #1
                  </a>
                </li>
                <li
                  data-test--id="navsubitem"
                >
                  <a
                    className="fr-nav__link"
                    target="_self"
                  >
                    Ressource #2
                  </a>
                </li>
                <li
                  data-test="navsubitem"
                >
                  <a
                    className="fr-nav__link"
                    target="_self"
                  >
                    Ressource #3
                  </a>
                </li>
              </ul>
            </div>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;
